snuba/api.py

```diff
@@ -344,37 +355,38 @@ def sdk_distribution(validated_body, timer):
 if application.debug or application.testing:
     # These should only be used for testing/debugging. Note that the database name
     # is checked to avoid scary production mishaps.
-    assert settings.CLICKHOUSE_TABLE in ('dev', 'test')
 
-    _ensured = False
+    _ensured = {}
 
-    def ensure_table_exists(force=False):
+    def ensure_table_exists(dataset, force=False, drop=False):
         global _ensured
 
-        if not force and _ensured:
+        if not force and _ensured.get(dataset, False):
             return
 
         from snuba.clickhouse import get_table_definition, get_test_engine
-
-        clickhouse_rw.execute(
-            get_table_definition(
-                name=settings.CLICKHOUSE_TABLE,
-                engine=get_test_engine(),
+        if not dataset.PROD:
```

Comment:
I think this should be solved by removing the PROD flag altogether. And it is not possible to call this from production since this function is only defined within this block:

`if application.debug or application.testing:`